en/components/query-builder.md

```diff
@@ -131,12 +143,44 @@ The `expressionTree` is a two-way bindable property which means a corresponding
 
 ```html
 <igx-query-builder #queryBuilder
-    [fields]="fields"
+    [entities]="entities"
     [(expressionTree)]="expressionTree"
     (expressionTreeChange)="onExpressionTreeChange()">
 </igx-query-builder>
 ```
 
+## Templating
+
+The Ignite UI for Angular Query Builder Component allows defining templates for header and search value using the following predefined reference names:
+
+### Header
+
+Passing content inside of the `igx-query-builder-header` allows templating the query builder header. The [`IgxQueryBuilderHeaderComponent`]({environment:angularApiUrl}/classes/igxquerybuilderheadercomponent.html) component provides a way to hide the legend by setting the [`showLegend`]({environment:angularApiUrl}/classes/igxquerybuilderheadercomponent.html#showLegend) input property is set to false.
+
+ The code snippet below illustrates how to do this:
+
+```html
+<igx-query-builder #queryBuilder [entities]="this.entities">
+    <igx-query-builder-header [showLegend]="false">
+        <span>Build your query</span>
+    </igx-query-builder-header>
+</igx-query-builder>
+```
+
+### Search value
+
+The search value of a condition can be templated using the [`igxQueryBuilderSearchValue`]({environment:angularApiUrl}/classes/igxquerybuildersearchvaluetemplatedirective.html) directive, applied to a `<ng-template>` inside of the `igx-query-builder`'s body:
+
+```html
+<igx-query-builder #queryBuilder [entities]="this.entities">
+    <ng-template #searchValueTemplate igxQueryBuilderSearchValue let-expression="expression">
+        <input type="text" required [(ngModel)]="expression.searchVal" />
```

Comment:
I suggest we use an IgxSelectComponent as we decided the sample to use it.

Reply:
Yes, this snippet will be updated once we have a templating sample ready.